rust/feature-flags/src/flags/flag_matching.rs

```diff
@@ -420,16 +421,42 @@ impl FeatureFlagMatcher {
             .initialize_group_type_mappings_if_needed(&feature_flags)
             .await;
 
-        // Evaluate all flags in the current level
-        let (flag_details_map, level_errors) = self
-            .evaluate_flags_in_level(
-                &feature_flags.flags,
-                &person_property_overrides,
-                &group_property_overrides,
-                hash_key_overrides,
-            )
-            .await;
-        errors_while_computing_flags |= level_errors;
+        let (flag_dependency_graph, errors) =
+            match DependencyGraph::from_nodes(&feature_flags.flags) {
+                Ok((graph, errors)) => (graph, errors),
+                Err(e) => {
+                    error!("Failed to build feature flag dependency graph: {:?}", e);
+                    return FlagsResponse {
+                        errors_while_computing_flags: true,
+                        flags: HashMap::new(),
+                        quota_limited: None,
+                        request_id,
+                        config: ConfigResponse::default(),
+                    };
+                }
+            };
+
+        if !errors.is_empty() {
+            errors_while_computing_flags = true;
+            // TODO: Make sure this call reports a useful error message.
+            error!("There were errors building the feature flag dependency graph. Will attempt to evaluate the rest of the flags: {:?}", errors);
+        }
```

Comment:
in both of these error paths, I would increment a metric counter (like [this](https://github.com/PostHog/posthog/blob/master/rust/feature-flags/src/flags/flag_service.rs#L58-L62) so we can scrape them on our [dashboard](https://grafana.prod-us.posthog.dev/d/eehu20mrbkbuof/new-feature-flags?var-reason=$__all&var-team_id=$__all&var-timing_quartile=0.99).  Each time I add metrics to this service I'm grateful I have them, it's super easy, and costs basically nothing at this point.